ext/handlers/filters/message/message.go

```diff
@@ -306,3 +306,15 @@ func TopicAction(msg *gotgbot.Message) bool {
 	return TopicEdited(msg) || TopicCreated(msg) ||
 		TopicClosed(msg) || TopicReopened(msg)
 }
+
+func MessageEntityPre(offset int, length int, language string) filters.Filter {
+	return func(ctx *gotgbot.CallbackContext) bool {
+		entities := ctx.Message.Entities
```

Comment:
This will only work on text messages, and not on media messages; that relies on message.CaptionEntities.

Reply:
about it only working on text messages and not media messages, yeah iam included media messages in below cod

`func MessageEntityPre(offset int, length int, language string) filters.Filter {
	return func(ctx *gotgbot.CallbackContext) bool {
		var entities []*gotgbot.MessageEntity
		if ctx.Message.Text != nil {
			entities = ctx.Message.Entities
		} else if ctx.Message.CaptionEntities != nil {
			entities = ctx.Message.CaptionEntities
		} else {
			return false // Neither text nor caption entities are present
		}

		for _, entity := range entities {
			if entity.Offset == offset && entity.Length == length && entity.Language == language {
				return true
			}
		}
		return false
	}
}
`

Reply:
Please don't make code changes in the PR comments, make your code changes to the PR itself. I cant merge a comment :)